### ISSUE TITLE
unused-ignore-comment will still flag itself as a warning

### ISSUE THREAD

**Issue by krpatter-intc** [astral-sh/ty#3393]
### Summary

I have some odd run-to-run issue where about 1 out of 5 times it reports an error with `inconsistent-mro` -- I think I'm hitting some flavor of this issue:

https://github.com/astral-sh/ty/issues/2052

But it does not fail that way every time, so if I added: `ty: ignore[inconsistent-mro]` I end up failing with `unused-ignore-comment`

But funnily enough if I add both:

`# ty: ignore [inconsistent-mro, unused-ignore-comment]`

Then when the error hits, I fail with the message in the title:

`warning[unused-ignore-comment]: Unused `ty: ignore` directive: 'unused-ignore-comment'`

So my underlying issue is really tied to #2052, but it feels that maybe `unused-ignore-comment` should get special treatment and not flag itself?

**Comment by MeGaGiGaGon** [comment]
You are getting some sort of non-deterministic diagnostic (#1670), since if everything was working correctly ty should always output the same thing on every run. If you could share the code that is causing the non-determinism, or a smaller reproduction of it, that would be appreciated.

Without knowing what your code is it's hard to say if #2052 is related, but it is a separate issue from the non-determinism.

**Comment by krpatter-intc** [comment]
Understood, I'll see what I can do for a small repro case of the inderminisitc, but for the fact that `unused-ignore-comment does` not work for itself, that seemed like perhaps a unique issue.  Having it honor itself would be helpful until the #1670 is resolved. For now, I used the pyproject.toml to suppress them through the whole project, but it would be nice to use just on the lines that are having the indeterministic issue.

```python
def temp(arg: int):
    return int(arg)+1

foo: str
foo = "1"
temp(foo) # ty: ignore [invalid-argument-type]
# both of these fail:
temp(foo) # ty: ignore [invalid-argument-type, unused-ignore-comment] 
temp(1) # ty: ignore [unused-ignore-comment]

```

**Comment by MichaReiser** [comment]
You can do this:

```py
temp(foo) # ty: ignore [invalid-argument-type, unused-ignore-comment, unused-ignore-comment] 
```

It looks a bit silly, but I do think warning about unused ignore comments by default is good.

**Comment by krpatter-intc** [comment]
```python
temp(foo) # ty: ignore [invalid-argument-type, unused-ignore-comment, unused-ignore-comment] 
```

Thanks. This works and it allows me to not ignore at the project level. I like having the check enabled, so I'll take silly if it lets me leave the setting on 😄 So perhaps it makes this issue just a low priority feature request to not need it twice. I _kindof_ get why it works now that I see it, but I dont think I would have ever figured that out.

**Comment by krpatter-intc** [comment]
Perhaps one more use that might be more common and so posting it here in case it is helpful to the issue or someone else runs in to this use case:

I manually run `uvx ty check` when doing development where I have all my deps installed. However, I also have it as a pre-commit hook with my `allow-unresolved-imports` filled in in the `pyproject.toml`

I have a dependency with a bug in their typing and I need to ignore the invalid argument type.

```
# ty: ignore [invalid-argument-type, unused-ignore-comment, unused-ignore-comment]
```

Using the 2x `unused-ignore-comment` is helpful there to allow both the manual run (with my deps) and the pre-commit style (without my deps) to run successfully.

**Comment by JulianJvn** [comment]
While I agree warning about unused ignore comments by default is good, having to use two `unused-ignore-comment` comments to silence a warning is unexpected to users. The `unused-ignore-comment` ignore comment already tells ty that the user wants to ignore the warning. Why should the user have to tell ty that they really want to ignore it by adding a second `unused-ignore-comment`? Even if we said requiring two ignore comments makes sense, why stop at two? Isn't one of the `unused-ignore-comment` ignore comments again unused, requiring an infinite number of ignore comments? I think it would be best if `unused-ignore-comment` ignore comments didn't trigger `unused-ignore-comment` warnings at all, as suggested in the issue description.

To explain my use case: I run CI workflows on code that uses `ctypes.WinDLL`. The code is used on Windows, but some CI workflows are run on Linux, and there is no `ctypes.WinDLL` there. ty rightfully issues an `unresolved-import` warning on Linux, but not on Windows. If I try to ignore the warning by adding an `unresolved-import` ignore comment, this leads to an `unused-ignore-comment` on Windows. But if I add an additional `unused-ignore-comment` ignore comment to ignore that, this triggers an `unused-ignore-comment` on that comment on Linux. Adding another `unused-ignore-comment` ignore comment fixes it, but seems unexpected as explained above.

**Comment by MichaReiser** [comment]
@JulianJvn I think https://docs.astral.sh/ty/reference/configuration/#replace-imports-with-any_1 might be interesting for you on linux

**Comment by JulianJvn** [comment]
Thanks. Sure, there are other ways to suppress the warning, but I'd prefer an ignore comment because it is near the location of the issue (the `import` line), and it will work more easily in other contexts, e.g., if someone checks out the code and runs ty manually. As far as I get it, there's no easy way to automatically apply the setting just on Linux, since I don't want the import to resolve to `Any` on Windows.

**Comment by MichaReiser** [comment]
That's correct. ty's configuration does not support conditional configs based on platform. This might be worth another issue. The only solution today would be to use a user-level configuration, but that defeats the purpose because it also requires a manual step.
